### PR TITLE
feat(dataimport): add delete_users management command

### PR DESCRIPTION
**Pull request by etnguyen03**
## Proposed changes
- Add the `delete_users` management command

## Brief description of rationale
To allow us to mass remove users from Ion with their FCPS student IDs, for instance, when they have withdrawn.

I will squash all commits before merging.

### CODE REVIEW

File: intranet/apps/dataimport/management/commands/delete_users.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        # Read in the list of student IDs, either from args or from the given filename
+        if options["student_ids"] is None:
+            ids = self.read_student_ids(options["filename"], options["header"])
+        else:
+            ids = options["student_ids"]
+
+        # Get a list of user model instances
+        users = []
+        for student_id in ids:
+            user = get_user_model().objects.filter(student_id=student_id).first()
```
Comment (Laur04) [inline intranet/apps/dataimport/management/commands/delete_users.py:64]: Why are you using `filter()` here and not `get()`? `student_id` is unique, so I think `get()` is more appropriate.

File: intranet/apps/dataimport/management/commands/delete_users.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            if user is not None:
+                users.append(user)
```
Comment (Laur04) [inline intranet/apps/dataimport/management/commands/delete_users.py:66]: You could replace this with:
```
if get_user_model().objects.get(student_id=student_id).exist():
  users.append(get_user_model().objects.get(student_id=student_id))
```

File: intranet/apps/dataimport/management/commands/delete_users.py (hunk 1 above)
Comment (etnguyen03) [reply intranet/apps/dataimport/management/commands/delete_users.py:64]: My reasoning was that `get()` would raise a `DoesNotExist` if the given user(s) do not exist, which isn't the case with the data we're working with (some given users were already deleted).

From what I've seen elsewhere, the two ways to deal with this problem are a try-catch or the `filter()` that I used.

Comment (abagali1) [reply intranet/apps/dataimport/management/commands/delete_users.py:64]: You can also do .filter(student_id__in=ids) which will just return a QuerySet of all the students who have a student_id in ids

Comment (abagali1) [reply intranet/apps/dataimport/management/commands/delete_users.py:64]: You can also do .filter(student_id__in=ids) which will just return a QuerySet of all the students who have a student_id in ids

Comment (Laur04) [reply intranet/apps/dataimport/management/commands/delete_users.py:64]: Yea that’s probably the best way to minimize queries.

**Comment by etnguyen03** [comment]
@Laur04?

File: intranet/apps/dataimport/management/commands/delete_users.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        if options["student_ids"] is None:
+            ids = self.read_student_ids(options["filename"], options["header"])
+        else:
+            ids = options["student_ids"]
+
+        # Get a list of user model instances
+        users = get_user_model().objects.filter(student_id__in=ids)
+
+        self.stdout.write(self.style.WARNING("These users are about to be deleted."))
+        self.stdout.write(str([u.username for u in users]))
```
Comment (theo-o) [inline intranet/apps/dataimport/management/commands/delete_users.py:65]:
```suggestion
        self.stdout.write(str([users.values_list("username", flat=True)]))
```
Minor performance nitpick

**Pull request merged**